### ISSUE TITLE
Headers in netlify.toml ignored by netlify dev

### ISSUE THREAD

**Issue by Tenga** [netlify/cli#1198]
**Describe the bug**

By reading [this documentation](https://docs.netlify.com/configure-builds/file-based-configuration/#sample-file), it seems like rules for headers should be able to be specified in `netlify.toml`.

However, `netlify dev` seems to only pull headers from `_headers` files and ignores `netlify.toml`.

Looking at the source code, it seems like this file/section deals with headers, and it only deals with `_headers` files, while not touching `netlify.toml`:

https://github.com/netlify/cli/blob/894ed447b1b93855c73f394db15a40b37d05b20a/src/utils/proxy.js#L229-L238

**To Reproduce**

1. Setup a project with headers in `netlify.toml` as described in [this documentation](https://docs.netlify.com/configure-builds/file-based-configuration/#sample-file)
2. Run `netlify dev`
3. Open your browser at the URL provided by `netlify.dev`
4. Inspect the headers on the requests
5. Observe that custom headers are not present
6. Add `out_publish/_headers` file with
```
/*
  X-Test: test
```
7. Observe that the CLI will refresh the headers, but just mention `_headers` file
8. Refresh the page in the browser
9. Observe that the custom headers from `_headers` are present

**Configuration**

<details>
  <summary>netlify.toml contents</summary>

```toml
[build]
    command   = "npm run build"
    functions = "out_functions"
    publish   = "out_publish"

[dev]
    functions = "out_functions"
    publish   = "out_publish"
    framework = "#static"

[[headers]]
    for = "/*"
    [headers.values]
        X-Test = "test"
```
</details>

<details>
  <summary>Environment info</summary>

```bash
System:
  OS: Windows 10 10.0.18362
  CPU: (6) x64 Intel(R) Core(TM) i5-8600K CPU @ 3.60GHz
  Memory: 7.11 GB / 15.95 GB
Binaries:
  Node: 12.18.3 - C:\Program Files\nodejs\node.EXE
  npm: 6.14.6 - C:\Program Files\nodejs\npm.CMD
```

The provided command doesn't give the Netlify CLI version, but `netlify -v` is:
```bash
netlify-cli/2.61.0 win32-x64 node-v12.18.3
```
</details>

**Expected behavior**

`netlify dev` should read and apply headers from `netlify.toml`

**Additional context**

Apologies, I don't have a cloneable minimal repro repository for you to clone at this point in time, it's my first foray into Netlify, and I'm using `next-on-netlify` based setup, I'm hoping the above repro is sufficient, if it is not, please let me know and I'll try to sort it out.

I've tried looking for known issues, duplicates or other gotchas, but so far I've run into nothing apart from the feeling that this is a bug by reading the source.

**Comment by jasikpark** [comment]
https://community.netlify.com/t/netlify-dev-not-loading-headers-from-netlify-toml/30068/2

**Comment by ehmicky** [comment]
Great news! This is being fixed at #3215.
